lib/resty/aes.lua

```diff
@@ -125,50 +125,75 @@ cipher = function (size, _cipher)
 end
 _M.cipher = cipher
 
-function _M.new(self, key, salt, _cipher, _hash, hash_rounds)
+
+local options = {
+    default = 0x01,
+    key_without_gen = 0x02
+    -- may add other things like padding mode; or use bit 'or/and' operation for combined option
+}
+_M.options = options
+
+function _M.new(self, key, salt, _cipher, _hash, hash_rounds, option)
     local encrypt_ctx = ffi_new(ctx_ptr_type)
     local decrypt_ctx = ffi_new(ctx_ptr_type)
     local _cipher = _cipher or cipher()
     local _hash = _hash or hash.md5
     local hash_rounds = hash_rounds or 1
+    local option = option or options.default
     local _cipherLength = _cipher.size/8
     local gen_key = ffi_new("unsigned char[?]",_cipherLength)
     local gen_iv = ffi_new("unsigned char[?]",_cipherLength)
 
-    if type(_hash) == "table" then
-        if not _hash.iv or #_hash.iv ~= 16 then
-          return nil, "bad iv"
-        end
+    if option == options.default then
+        if type(_hash) == "table" then
+            if not _hash.iv or #_hash.iv ~= 16 then
+              return nil, "bad iv"
```

Comment:
We should use 4-space indentations consistently. Please fix other similar places, if any.